var/spack/repos/builtin/packages/wgl/package.py

```diff
@@ -0,0 +1,51 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import re
+import sys
+
+from spack.package import *
+
+
+class Wgl(BundlePackage):
+    """External WGl and Windows OpenGL emulation representation in Spack"""
+
+    homepage = "https://learn.microsoft.com/en-us/windows/win32/opengl/wgl-and-windows-reference"
+
+    # hard code the extension as shared lib
+    libraries = ["opengl32.dll"]
+
+    # As per https://github.com/spack/spack/pull/31748 this version and it's provisory represent
+    # an arbitrary openGL version designed for maximum compatibility with calling packages
+    # this current version simply reflects the latest OpenGL vesion available at the time of
+    # package creation and is set in a way that all specs currently depending on GL are satisfied appropriately
+    version("4.6")
+
+    provides("gl@4.6")
+
+    # WGL exists on all Windows systems post win 98, however the headers
+    # needed to use OpenGL are found in the SDK (GL/gl.h)
+    # Dep is needed to consolidate sdk version to locate header files for
+    # version of SDK being used
+    depends_on("win-sdk")
+
+    # WGL has no meaning on other platforms, should not be able to spec
+    for plat in ["linux", "darwin", "cray"]:
+        conflicts("platform=%s" % plat)
+
+
+    def determine_version(self, exe):
+        """Allow for WGL to be externally detectable"""
```

Comment:
I assume there would be a problem if this returns `None`: is there a way to get a version from the file? Also, this calls the file `exe` when in fact the class defines `libraries` (not strictly an error, but not idiomatic).

Reply:
I notice other package implementations of `determine_version` don't raise errors on failure, is there a particular reason we want to here? I'm perfectly fine with including it, I'd just like to understand.

Reply:
Sorry: I was wrong about this - it's totally fine to return none (in that case, the object under consideration does not produce an associated spec).